src/components/popups/profile-settings/EmailForm.tsx

```diff
@@ -108,11 +109,14 @@ export default function EditEmail({ show, onClose }: EditProfileProps): ReactEle
                   value: 2,
                   message: "This must be a valid email address",
                 },
+                validate: (val: string) => {
+                  if (watch('email') != val) return "Emails should match."
```

Comment:
Strictly comparison is more reliable.

```suggestion
                  if (watch('email') !== val) return "Emails should match."
```